appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/Padding.kt

```diff
@@ -0,0 +1,105 @@
+package com.bumble.appyx.interactions.core.ui.property.impl
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.AnimationVector4D
+import androidx.compose.animation.core.Easing
+import androidx.compose.animation.core.TwoWayConverter
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.lerp
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.property.Interpolatable
+import com.bumble.appyx.interactions.core.ui.property.MotionProperty
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class Padding(
+    uiContext: UiContext,
+    target: Target,
+    displacement: StateFlow<Target> = MutableStateFlow(Target()),
+    visibilityThreshold: Target = Target(1.dp, 1.dp, 1.dp, 1.dp),
+) : MotionProperty<Padding.Target, AnimationVector4D>(
+    uiContext = uiContext,
+    animatable = Animatable(target, Target.VectorConverter),
+    easing = target.easing,
+    visibilityThreshold = visibilityThreshold,
+    displacement = displacement
+), Interpolatable<Padding.Target> {
+
+    class Target(
+        val top: Dp = 0.dp,
+        val bottom: Dp = 0.dp,
+        val start: Dp = 0.dp,
+        val end: Dp = 0.dp,
```

Comment:
funny that @zsoltk asked to do exactly the opposite, @zsoltk wdyt?

Reply:
The actual issue lies in how the generated code implements the animateTo method. It's something similar to the following Kotlin code:

```kotlin
override suspend fun animateTo(
  scope: CoroutineScope,
  target: TargetUiState,
  springSpec: SpringSpec<Float>,
): Unit {
  listOf(
    scope.async {
      padding.animateTo(
        target.padding.value, // <-- Error: unresolved reference
        spring(springSpec.dampingRatio, springSpec.stiffness),
      )
    },
  ).awaitAll()
}
```

In the KSP processor, we can write a special case for Padding and provide all the resolvable references (`top, bottom, left, right`), but we will still have an issue here because the `animateTo` method from `MotionProperty<T, V>` only accepts a single parameter from the `Target`. So, I believe we still need to keep the redundancy you mentioned.